<a href="https://colab.research.google.com/github/sheinviktoriia/Data-analyst-home-works-/blob/main/HW_1_1_3_%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D0%B8_%D1%80%D0%BE%D0%B1%D0%BE%D1%82%D0%B8_%D0%B7_%D1%80%D1%8F%D0%B4%D0%BA%D0%B0%D0%BC%D0%B8_%D1%82%D0%B0_%D0%BC%D0%B0%D1%82%D0%B5%D0%BC%D0%B0%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D0%BE%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D1%96%D1%97_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

In [2]:
print((a + b) * c / d)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [3]:
number_one = 11283
number_two = 89
number_one % number_two

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [4]:
number_one // number_two

126

4. Запишіть у змінну `name` ваше імʼя.

In [5]:
name = 'Viktoriia'

5. Запишіть у змінну `surname` ваше прізвище.

In [6]:
surname = 'Shein'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [7]:
name + ' ' + surname

'Viktoriia Shein'

In [8]:
name_surname = name + ' ' + surname
print(name_surname)

Viktoriia Shein


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [9]:
name_surname.upper()

'VIKTORIIA SHEIN'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [10]:
char = name_surname

In [11]:
char[2]

'k'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [15]:
char[10], char[14]

('S', 'n')

In [16]:
surname[0], surname[4]

('S', 'n')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [17]:
len(surname)

5

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [44]:
M = 'Antananarivo'
M.count('A') + M.count('a')

4

Це завдання зайняло 30 хв і я не впевнена, що це найзручніший варіант визначення.

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [46]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'
my_string.startswith('The')
my_string.endswith('dog!')

False

In [47]:
my_string.startswith('The')

True

In [48]:
my_string.endswith('dog!')

False

13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [50]:
print(my_string.split())

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']


In [52]:
print('_'.join(my_string))

T_h_e_ _q_u_i_c_k_ _b_r_o_w_n_ _f_o_x_ _j_u_m_p_s_ _o_v_e_r_ _t_h_e_ _l_a_z_y_ _d_o_g_!_!_!


In [53]:
print('_'.join(['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']))

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`